infra/app/env-template/main.tf

```
@@ -117,3 +118,24 @@ module "analytics" {
   subnet_ids           = data.aws_subnets.default.ids
   service_cluster_arn  = module.service_cluster.service_cluster_arn
 }
+
+data "aws_iam_role" "participant_task_executor" {
+  # Referencing the task executor of the ECS services so that they have the ability to upload documents to s3
+  name = "${local.participant_service_name}-task-executor"
+
+}
+
+data "aws_iam_role" "staff_task_executor" {
+  # Referencing the task executor of the ECS services so that they have the ability to upload documents to s3
+  name = "${local.staff_service_name}-task-executor"
+}
+
+module "doc_upload" {
+  source           = "../../modules/s3-encrypted"
+  environment_name = var.environment_name
+  s3_bucket_name   = local.document_upload_s3_name
+  # read_role_arns   = [data.aws_iam_role.staff_task_executor.arn, data.aws_iam_role.participant_task_executor.arn]
+  read_role_names   = [data.aws_iam_role.staff_task_executor.name, data.aws_iam_role.participant_task_executor.name]
+  write_role_names  = [data.aws_iam_role.staff_task_executor.name, data.aws_iam_role.participant_task_executor.name]
```

Only the Remix app needs to be able to write.
```suggestion
  write_role_names  = [data.aws_iam_role.participant_task_executor.name]
```